tests/JestExt.test.ts

```diff
@@ -810,4 +828,28 @@ describe('JestExt', () => {
       expect(channelStub.appendLine).not.toHaveBeenCalled();
     });
   });
+  describe('_updateCoverageMap', () => {
+    it('the overlay and codeLens will be updated when map updated async', async () => {
+      expect.hasAssertions();
+      (CoverageMapProvider as jest.Mock<any>).mockImplementation(() => ({
+        update: () => Promise.resolve(),
+      }));
+      const coverageCodeLensProvider: any = { coverageChanged: jest.fn() };
+      const sut = new JestExt(
+        null,
+        workspaceFolder,
+        projectWorkspace,
+        channelStub,
+        extensionSettings,
+        debugCodeLensProvider,
+        debugConfigurationProvider,
+        null,
+        null,
+        coverageCodeLensProvider
+      );
+      await sut._updateCoverageMap({});
```

Comment:
**Thought experiment**: Take a moment to reflect on how we can test this behaviour without testing the implementation details of our private functions. I've been reflecting on that a lot more lately. Can we keep the tests stable as we refactor? 😉

Leave this as is though. 👍

Reply:
Agree, I really don't like how `JestExt` is set up for testing, it is difficult to test so we end up either no testing it or had to break it into smaller testable pieces that blur the line of public vs private functions. We should refactor this  in the future...